lsp-mode.el

```diff
@@ -3256,14 +3264,16 @@ disappearing, unset all the variables related to it."
                            lsp--registered-capability-options
                            (lsp:did-change-watched-files-registration-options-watchers)
                            (seq-find
-                            (-lambda ((&FileSystemWatcher :glob-pattern))
-                              (-let [glob-regex (lsp-glob-to-regexp glob-pattern)]
-                                (or (string-match glob-regex changed-file)
-                                    (string-match glob-regex (f-relative changed-file root-folder)))))))))))
+                            (-lambda ((&FileSystemWatcher :glob-pattern :kind?))
+                              (and (or (null kind?)
+                                       (> (logand kind? watch-bit) 0))
```

Comment:
nit: you may use natnump instead of `> 0`

Reply:
I don't think `natnump` works here since we want to reject 0 as well, and 0 is a natural number.